pkg/syncer/util.go

```diff
@@ -707,8 +707,18 @@ func createCnsVolume(ctx context.Context, pv *v1.PersistentVolume,
 	} else {
 		log.Infof("vSphere CSI Driver has successfully marked volume: %q as the container volume.",
 			pv.Spec.CSI.VolumeHandle)
-
-		if isMultiVCenterFssEnabled && len(metadataSyncer.configInfo.Cfg.VirtualCenter) > 1 {
+		if metadataSyncer.clusterFlavor == cnstypes.CnsClusterFlavorVanilla {
+			if isMultiVCenterFssEnabled && len(metadataSyncer.configInfo.Cfg.VirtualCenter) > 1 {
+				// Create CNSVolumeInfo CR for the volume ID.
+				err = volumeInfoService.CreateVolumeInfo(ctx, pv.Spec.CSI.VolumeHandle, vcHost)
+				if err != nil {
+					log.Errorf("Failed to store volumeID %q for vCenter %q in CNSVolumeInfo CR. Error: %+v",
+						pv.Spec.CSI.VolumeHandle, vcHost, err)
+					return err
+				}
+			}
+		} else if metadataSyncer.clusterFlavor == cnstypes.CnsClusterFlavorWorkload &&
+			commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.PodVMOnStretchedSupervisor) {
 			// Create CNSVolumeInfo CR for the volume ID.
 			err = volumeInfoService.CreateVolumeInfo(ctx, pv.Spec.CSI.VolumeHandle, vcHost)
```

Comment:
Shouldn't this be `CreateVolumeInfoWithPolicyInfo` instead of `CreateVolumeInfo`?

Reply:
I think CreateVolumeInfo needs to be updated to take PolicyID as parameter.

Reply:
In static provisioning case, I haven't checked if we can fetch storage policy id & name. Taking a look ...